src/Features/Core/Portable/ReplaceMethodWithProperty/AbstractReplaceMethodWithPropertyService.cs

```diff
@@ -47,18 +47,35 @@ private static bool OverridesMetadataSymbol(IMethodSymbol method)
         {
             var getMethodDeclaration = getAndSetMethods.GetMethodDeclaration;
             var setMethodDeclaration = getAndSetMethods.SetMethodDeclaration;
-            if (setMethodDeclaration == null)
+            var finalLeadingTrivia = getAndSetMethods.GetMethodDeclaration.GetLeadingTrivia().ToList();
+
+            //If there is a comment on the same line as the method it is contained in trailing trivia for the parameter list
+            //If it's there we need to add it to the final comments
+            //this is to fix issue 42699, https://github.com/dotnet/roslyn/issues/42699
+            if (getMethodDeclaration.ChildNodes().Any(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList && n.GetTrailingTrivia().Any(t => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t))))
             {
-                return property.WithLeadingTrivia(getMethodDeclaration.GetLeadingTrivia());
+                //we have a meaningful comment on the parameter list so add it to the trivia list
+                finalLeadingTrivia.AddRange(getMethodDeclaration.ChildNodes().Where(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList).First().GetTrailingTrivia());
             }
 
-            var finalLeadingTrivia = getAndSetMethods.GetMethodDeclaration.GetLeadingTrivia().ToList();
+            if (setMethodDeclaration == null)
+            {
+                return property.WithLeadingTrivia(finalLeadingTrivia);
+            }
 
             finalLeadingTrivia.AddRange(
                 setMethodDeclaration.GetLeadingTrivia()
                                     .SkipWhile(t => syntaxFacts.IsEndOfLineTrivia(t))
                                     .Where(t => !t.IsDirective));
 
+            //If there is a comment on the same line as the method it is contained in trailing trivia for the parameter list
+            //If it's there we need to add it to the final comments
+            if (setMethodDeclaration.ChildNodes().Any(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList && n.GetTrailingTrivia().Any(t => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t))))
+            {
+                //we have a meaningful comment on the parameter list so add it to the trivia list
+                finalLeadingTrivia.AddRange(setMethodDeclaration.ChildNodes().Where(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList).First().GetTrailingTrivia());
+            }
+
```

Comment:
Should extract this to a method, as its duplicated with the above, except operates on a different method declaration.

Reply:
I pushed changes with the code extracted to a separate function.